### PR TITLE
Redesign/revision database

### LINKED ISSUE #67: Edit button/link text component broken
It looks like certain  elements of the update/create link modal are broken:
- Currently I cannot change the colour of links
- The preview doesn't seem to style correctly, though this may be in part due to the above issue
- We should probably look at trimming the text displayed in a link/button

image

### LINKED ISSUE #71: Remove attribute isButton from CmsButtons
#70  means we no longer need hyperlinks

### PR DESCRIPTION
**Pull request by jacksos101**
This pull request makes some big changes to the way that revisions are stored in the database. It also changes the front-end design for when users edit pages.

Key changes:

- Users now provide a reason (optional) for page revisions
- Buttons now have a sharper, more modern style
- Add CKEditor for text field revisions

Bugs to fix:
- [x] Deserializing JSON string in RevisionController.CreateRevision() fails, meaning we can't actually update the page anymore

- [x] Button configuration dropdown does not move to the correct location after deleting a button.

- [x] Update template.json and BasePage.js so that page template IDs start at 1, rather than 0

fixes #67 
fixes #71

### CODE REVIEW

File: Data/Seeds/templates.json [hunk 1]
```diff
@@ -1,8 +1,16 @@
 [
   {
+    "Id":  0,
     "Name": "Contact Us",
     "Description": "A contact us layout that features three text areas, an email form and Google Maps integration.",
```
Comment (samuelgrant) [inline Data/Seeds/templates.json]: As per my discussion with @jacksos101  the Id field should start at one and go up from there

File: Resources/js/React/PageTemplates/BasePage.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
+﻿import React, { Component, Fragment } from 'react';
+import { render } from 'react-dom';
+import PageControls from './PageControls';
+import ReactTemplate1 from './1';
+import ReactTemplate2 from './2';
+import $ from 'jQuery';
+
+const baseUri = `/api/pages`;
+const components = {
+    0: ReactTemplate1,
```
Comment (samuelgrant) [inline Resources/js/React/PageTemplates/BasePage.js]: These numbers need to start at 1 and match the `templates.json` file

Comment (jacksos101) [reply Resources/js/React/PageTemplates/BasePage.js]: This has now been done.

**Comment by samuelgrant** [comment]
@jacksos101 

> I believe this PR is good to go.
> We should, in the future, determine a way to disable the publish button if there are pending edits.
> We may also want to disable multiple concurrent edits of text components.

**Pull request merged**